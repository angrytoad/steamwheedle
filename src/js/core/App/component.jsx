// @flow
import React from 'react';
import { Switch, Route } from 'react-router';
import {ToastContainer, ToastStore} from 'react-toasts';
import type { AppProps } from './container';
import '../../../styles/core.scss';
import TestActions from '../../store/actions/test.actions';

import background from "../../../resources/images/steamwheedle_background.jpg";
import css from "./styles.module.scss";

import Landing from "../Landing/container";
import NoRouteMatch from "../NoRouteMatch/component";
import Register from "../Register/container";

class App extends React.PureComponent<AppProps> {
  testActions = new TestActions();

  componentDidMount = () => {
    //this.testActions.testAction();
  };

  renderPathComponent = () => {
    return (
      <Switch>
        <Route exact path="/" component={Landing} />
        <Route exact path="/register" component={Register} />
        <Route component={NoRouteMatch} />
      </Switch>
    )
  };

  render() {
    return (
      <div
        className={css.App}
        style={{
          backgroundImage: `url(${background})`,
        }}
      >
        <div
          className={css.backDrop}
        />
        <ToastContainer store={ToastStore}/>
        { this.renderPathComponent() }
      </div>
    );
  }
}

export default App;
