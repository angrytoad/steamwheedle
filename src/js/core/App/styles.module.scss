@import "../../../styles/variables";

.App{
  position: relative;
  color:$textLight;
  text-align: center;
  min-height: 100vh;
  display:grid;
  align-items: center;
  background-size:cover;
  background-repeat: no-repeat;
  z-index: 0;

  > div{
    padding: 1rem;
  }

  .backDrop{
    background-color: rgba($darkBackground,0.3);
    position: absolute;
    top:0;
    bottom:0;
    right:0;
    left:0;
    z-index: -1;
  }

  .landing{
    position: relative;
    z-index: 2;
  }

  .goldBar{
    background-repeat: repeat-x;
    height: 16px;
    background-position: center;
    width:100%;
    max-width: 520px;
    margin: auto;
  }



  .madeWithLove{
    margin:auto;
    max-width:600px;
    width:100%;

    ul{
      padding:0;
      margin:0;
      list-style-type: none;
      display:flex;
      justify-content: center;

      a{
        flex:1;
        max-width:80px;
        margin:1rem;

        img{
          width:100%;
        }
      }


    }
  }
}
