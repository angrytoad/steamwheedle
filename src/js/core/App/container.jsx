// @flow
import connect from 'react-redux/es/connect/connect';
import { withRouter } from 'react-router';
import type { AppState } from '../../store/state';
import App from './component';
import history from "../../routing/history";

type ReduxStateProps = {|

|};

type ReduxActionProps = {|

|};

export type AppProps = {
  ...ReduxStateProps,
  ...ReduxActionProps,
  history: history,
};

const mapStateToProps = (state: AppState): ReduxStateProps => ({
  test: state.test,
});

export default withRouter(connect(mapStateToProps)(App));
