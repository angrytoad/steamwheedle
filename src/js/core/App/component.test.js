import React from 'react';
import { compose } from 'redux';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './component';

const shallowExpect = compose(
  expect,
  toJson,
  shallow,
);


describe('App', () => {
  const props = {};

  it('Should match the snapshot', () => {
    shallowExpect(<App {...props} />).toMatchSnapshot();
  });
});
