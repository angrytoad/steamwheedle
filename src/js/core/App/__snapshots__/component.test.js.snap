// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match the snapshot 1`] = `
<div
  className="App"
>
  <h1>
    Hello
  </h1>
</div>
`;
